engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsANode.java

```diff
@@ -0,0 +1,82 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.AcceptsError;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+
+@BuiltinMethod(
+    type = "Meta",
+    name = "is_a",
+    description = "Checks type of a node.",
+    autoRegister = false)
+public abstract class IsANode extends Node {
+
+  public abstract boolean execute(@AcceptsError Object value, Object type);
+
+  public static IsANode build() {
+    return IsANodeGen.create();
+  }
+
+  boolean isAnyType(Type t) {
+    var ctx = EnsoContext.get(this);
+    return ctx.getBuiltins().any() == t;
+  }
+
+  @Specialization(guards = "isAnyType(anyType)")
+  boolean doAnyCheck(Object any, Type anyType) {
+    return true;
+  }
+
+  @Specialization
+  boolean doLongCheck(long value, Type type) {
+    return isNumberType(type);
+  }
+
+  @Specialization
+  boolean doDoubleCheck(double value, Type type) {
+    return isNumberType(type);
+  }
+
+  @Specialization
+  boolean doBigIntegerCheck(EnsoBigInteger value, Type type) {
+    return isNumberType(type);
+  }
```

Comment:
Would it make sense to add some caching to these specializations? Something like in `ExecuteCallNode`?

Reply:
What shall we cache?

Reply:
The `type` and result of `isNumberType` called on it? I'm not certain if this will help, but looking at similar usages it seems that it may work better if we run a check many times with the exact same type reference on the right (which seems like a common use-case).